corehq/util/dates.py

```diff
@@ -52,3 +53,31 @@ def safe_strftime(val, fmt):
     return safe_val.strftime(fmt
                              .replace("%Y", str(val.year))
                              .replace("%y", str(val.year)[-2:]))
+
+
+_assert = soft_assert('droberts' + '@' + 'dimagi.com')
+
+
+def iso_string_to_datetime(iso_string):
+    """
+    parse datetime string in iso format with or without microseconds,
+    always with both date and time
+    and always with the 'Z' UTC timezone suffix
+
+    return an offset-naive datetime representing UTC
+
+
+    >>> iso_string_to_datetime('2015-04-07T19:07:55Z')
+    datetime.datetime(2015, 4, 7, 19, 7, 55)
+    >>> iso_string_to_datetime('2015-04-07T19:07:55.437086Z')
+    datetime.datetime(2015, 4, 7, 19, 7, 55, 437086)
```

Comment:
Even just this quick doctest was able to catch a silly typo in my code. Yay tests!